index.js

```diff
@@ -83,20 +83,28 @@ module.exports = function (program, args, cb) {
   return child
 }
 
-function proxySignals (child) {
+/**
+ * Starts forwarding signals to `child` through `parent`.
+ *
+ * @param parent Parent process.
+ * @param child Child Process.
+ * @return `unproxy` function to stop the forwarding.
+ * @internal
+ */
+function proxySignals (parent, child) {
   var listeners = {}
   signalExit.signals().forEach(function (sig) {
     listeners[sig] = function () {
       child.kill(sig)
     }
-    process.on(sig, listeners[sig])
+    parent.on(sig, listeners[sig])
   })
 
   return unproxySignals
```

Comment:
Why not just return the function (with it's definition, could be named or anonymous):

```javascript
return function () {
  // ...
}
```

Reply:
The focus of this PR was to pass the parent as input so I kept the other changes minimal. I can change it if you want to.